### ISSUE TITLE
Allow passing aditional information to result-options plugins

### ISSUE THREAD

**Issue by tmgodinho** [bioinformatics-ua/dicoogle#206]
Lately I felt the need to access these attributes related to the row of the result-option plugin.

Namely I need to access the following fields:
- Modality
- Number of Images
- URI's of the images enclosed by the row - example all the images from the patient if the current level is the patient lvl.

Preferably, the plug-in should have access to all the DIM fields as well as an indication of the current selected field.

@bastiao 
@Enet4

**Comment by Enet4** [comment]
Well, I haven't though all the way through about the actual events, but they must, however, be triggered at least after the plugins are loaded. We might have to make the plugin itself signal the slot when it is ready to receive the contents... This is so awkwardly resembling that of the function callback mechanism that I wonder if we couldn't just expose a function-based API to achieve just that. Ideally, we'd only need to do something like this:

```javascript
slot.getCurrentResultData((entry) => {
  if (entry['Modality'] === 'MS') { /* ... */ }
});
```

This may be feasible and even more elegant than exchanging signals (which are not even limite to plugin-slot communication, but _all-plugins_-slot communication).

**Comment by Enet4** [comment]
This should be enough for @tmgodinho in the mean time: [imp/webapp/inject-more](https://github.com/bioinformatics-ua/dicoogle/tree/imp/webapp/inject-more)

- Added "data-result-modality" attribute in all levels (comma separated when there's more than one)
- Added "data-result-nimages", "data-result-nseries" and "data-result-nstudies" in places where such calculation is possible.
- Added "data-result-uris" at series level, pipe separated (eg. "file:/1/2.dcm|file:/1/3.dcm")

**Comment by Enet4** [comment]
I believe it's resolved in #322.
